harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Aligner.java

```diff
@@ -631,4 +622,43 @@ private Element processMetadata(ServiceContext context,
         }
         return md;
     }
+
+    /**
+     * Retrieves the list of metadata uuids that have the same dataset identifier.
+     *
+     * @param searchMan         ES search manager.
+     * @param datasetIdCode     Dataset identifier.
+     * @param metadataUuid      Metadata identifier to exclude from the search.
+     * @return  A list of metadata uuids that have the same dataset identifier.
+     */
+    private Set<String> retrieveMetadataUuidsFromIdentifier(EsSearchManager searchMan,
+                                                            String datasetIdCode,
+                                                            String metadataUuid) {
+
+         Set<String> metadataUuids = new HashSet<>();
+
+        String jsonQuery = " {" +
+            "       \"query_string\": {" +
+            "       \"query\": \"+resourceIdentifier.code:%s -uuid:%s\"" +
```

Comment:
Maybe we should do an exact match with `"` for both fields too even if this should work in most case probably?
Like in https://github.com/geonetwork/core-geonetwork/blob/main/services/src/main/java/org/fao/geonet/api/records/MetadataUtils.java#L252

Reply:
Updated with the suggested change.